star-rx72n-firmware/lib/rx_fec/inc/rx_fec.h

```diff
@@ -0,0 +1,265 @@
+                             size_t survivors_len);
+
+/**
+ * @brief Deinitialize FEC decoder
+ *
+ * @param[in,out] dec Pointer to decoder handle
+ * @return RX_OK on success
+ */
+rx_err_t rx_fec_decoder_deinit(rx_fec_decoder_t *dec);
+
+/**
+ * @brief Decode soft bits using Viterbi algorithm
+ *
+ * Performs soft-decision Viterbi decoding on received soft bits.
+ * Soft bits should be in pairs (G1, G2) for each encoded symbol.
+ *
+ * @param[in]  dec                 Initialized decoder handle
+ * @param[in]  soft_bits           Received soft bits (pairs)
+ * @param[in]  soft_len            Number of soft bits (must be even)
+ * @param[in]  expected_output_len Expected decoded length in bytes
+ * @param[out] output              Decoded output buffer
+ * @param[out] output_len          Actual decoded length
+ *
+ * @return RX_OK on success
+ * @return RX_ERR_INVALID_ARG if any pointer is NULL or soft_len is odd
+ * @return RX_ERR_INVALID_STATE if decoder not initialized
+ * @return RX_ERR_INVALID_SIZE if output buffer too small
+ */
+rx_err_t rx_fec_decode_soft(rx_fec_decoder_t *dec,
+                            const rx_soft_bit_t *soft_bits, size_t soft_len,
+                            size_t expected_output_len, uint8_t *output,
+                            size_t *output_len);
+
+/**
+ * @brief Decode hard bits using Viterbi algorithm
+ *
+ * Converts hard bits to soft bits internally, then performs Viterbi decoding.
+ * Less accurate than soft-decision decoding but useful for testing.
+ *
+ * @param[in]  dec                 Initialized decoder handle
+ * @param[in]  data                Received hard bits (packed bytes)
+ * @param[in]  data_len            Number of bytes
+ * @param[in]  expected_output_len Expected decoded length in bytes
+ * @param[out] output              Decoded output buffer
+ * @param[out] output_len          Actual decoded length
+ *
+ * @return RX_OK on success
+ * @return RX_ERR_INVALID_ARG if any pointer is NULL
+ * @return RX_ERR_INVALID_STATE if decoder not initialized
+ */
+rx_err_t rx_fec_decode_hard(rx_fec_decoder_t *dec, const uint8_t *data,
+                            size_t data_len, size_t expected_output_len,
+                            uint8_t *output, size_t *output_len);
+
+/* =============================================================================
+ * Utility Functions
+ * =============================================================================
+ */
+
+/**
+ * @brief Convert hard bit to soft bit
+ *
+ * @param[in] bit Hard bit value (0 or 1)
+ * @return Soft bit: k_soft_bit_min for 0, k_soft_bit_max for 1
+ */
+static inline rx_soft_bit_t rx_fec_hard_to_soft(uint8_t bit)
+{
+    return (bit != 0) ? k_soft_bit_max : k_soft_bit_min;
+}
+
+/**
+ * @brief Convert soft bit to hard bit
+ *
+ * @param[in] soft Soft bit value
+ * @return Hard bit: 1 if soft >= 0, 0 otherwise
+ */
+static inline uint8_t rx_fec_soft_to_hard(rx_soft_bit_t soft)
+{
+    return (soft >= 0) ? 1 : 0;
+}
+
```

Comment:
We should not have static methods inside header files, if there is a good reason for this please justify it with a comment in Github and also leave a comment here. When we use static methods inside headers it results in a larger binary and often its not needed. Usually, we should be placing the static methods inside the source `.c` file instead. Also refer to the style guide for understanding the prefix `internal_` and `priv_`. If this is meant to be exposed and not actually static use `priv_` but if its supposed to be static for only this `.c` then use `internal_` and place this static method inside the source.